src/lib/commandline_cmds.ts

```diff
@@ -155,6 +155,39 @@ export function getCommandlineFns(cmdline_state) {
             return messageOwnTab("controller_content", "acceptExCmd", [command])
         },
 
+        // Execute the excmd parameter on the completion args.
+        // Ignores the completion func.
+        execute_ex_on_completion: (excmd: string) => {
+            const command = cmdline_state.getCompletion()
+
+            cmdline_state.fns.hide_and_clear()
+
+            if (command === undefined) return
+
+            const [func, ...args] = command.trim().split(/\s+/)
+
+            if (func.length === 0 || func.startsWith("#") || args.length === 0) {
+                return
+            }
+
+            // Create new cmd to execute excmd with the completion
+            // args, discarding the completion func
+            const cmdToExec = excmd + " " + args.join(" ")
+
+            // Save non-secret commandlines to the history.
+            if (
+                !browser.extension.inIncognitoContext &&
+                !(excmd === "winopen" && args[0] === "-private")
```

Comment:
I'm wondering how this deals with `composite winopen -private` - I think usually we just discard anything that contains "winopen -private" but I might be wrong.

Reply:
> I _think_ the command line should probably persist rather than being closed. What do you think?

Yes, I agree.  Can you please advise regarding the best approach to implement a persistent cmd line?  It seems that even without `cmdline_state.fns.hide_and_clear()`, the cmdline hides on `return ownMessageTab()`.

Also, I noticed <S-Del> highlighted tab is not working yet in the All Tabs buffer (B).  I will fix this in the next commit.

Thanks for the feedback and help.

Reply:
I'm honestly not sure what we should do with `accept_line`. We should have the same behaviour everywhere. That probably means we should have a special "shouldStoreExString" function that they both use. I'm trying to work out if it's more annoying to have e.g. a search query that includes winopen -private disappear or have a sensitive composite command stored.

I think as part of this PR the current behaviour should be kept but the logic should be separated into its own function in e.g. src/lib/, so that if we change our minds we remember to change it everywhere. Sorry for changing my mind :)

I'll look into the command line hiding.

Reply:
The latest commit adds an a "store_ex_string" function to avoid code duplication, making it easier to change the logic if we change our minds.